### PR TITLE
Feat: Added tooltip for special unlock conditions in associated outputs.

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

- Added a tooltip for Storage Deposit Return Unlock Condition - `arrow_back` icon will be shown for outputs which has SDRUC and on hover you can see the `Return Amount`.
- Added a tooltip for Expiration Unlock Condition - `hour_glass` icon will be shown for outputs which has EUC and on hover you can see the `Time left for expiration`.
- Added a tooltip for Timelock Unlock Condition - `clock` icon will be shown for outputs which has EUC and on hover you can see the `Time left for locked output`.

Aims to complete https://github.com/iotaledger/explorer/issues/447

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

In Associated Outputs Table on Address page for the outputs which has special unlock conditions like (SDRUC, EUC or TUC) you can see an icon beside copy icon in `Output Type` column and on hover on that icon you can see the data.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by msarcev** [comment]
image
I think it will look better if we have these icons "stay" in-line with the copy icon when expanded.
I think for this we will need to push the code inside `Output.tsx`. But maybe it's even better that way cause we will get this feature everywhere Output is used.
Also maybe try to color them yellow ? See how it looks.

**Comment by abbasabid** [comment]
> image
> 
> I think it will look better if we have these icons "stay" in-line with the copy icon when expanded. I think for this we will need to push the code inside `Output.tsx`. But maybe it's even better that way cause we will get this feature everywhere Output is used. Also maybe try to color them yellow ? See how it looks.

Done.

**Review by msarcev — CHANGES_REQUESTED** [review]

File: client/src/app/components/stardust/Output.tsx [hunk 1]
```diff
@@ -47,17 +48,32 @@ class Output extends Component<OutputProps, OutputState> {
 
         this.state = {
             isExpanded: this.props.isPreExpanded ?? false,
-            isFormattedBalance: true
+            isFormattedBalance: true,
+            isSpecialCondition: false
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/app/components/stardust/Output.tsx]: Done.

File: client/src/helpers/stardust/nameHelper.ts [hunk 2]
```diff
@@ -29,17 +29,17 @@ export class NameHelper {
     public static getOutputTypeName(type: number): string {
         switch (type) {
             case BASIC_OUTPUT_TYPE:
-                return "Basic Output";
+                return "Basic";
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/helpers/stardust/nameHelper.ts:32]: Luca suggested that here https://github.com/iotaledger/explorer/issues/447#issuecomment-1231476297

File: client/src/app/components/stardust/AssociatedOutput.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ const AssociatedOutput: React.FC<AssociatedOutputProps> = ({ network, associated
 
     const outputTableRow = (
         <tr>
-            <td className="card">
+            <td className="card overflow-unset">
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/app/components/stardust/AssociatedOutput.tsx:62]: Yes

File: client/src/app/components/Tooltip.scss [hunk 4]
```diff
@@ -3,7 +3,6 @@
 .tooltip {
     position: relative;
     display: flex;
-    align-items: center;
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/app/components/Tooltip.scss:6]: The tooltip is using in three places and i checked that they look fine.

File: client/src/app/components/stardust/Output.scss [hunk 5]
```diff
@@ -46,6 +47,11 @@
             }
         }
     }
+
+    .icon {
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/app/components/stardust/Output.scss]: Done.

File: client/src/app/components/stardust/AssociatedOutput.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable jsdoc/require-param */
 /* eslint-disable jsdoc/require-returns */
 import moment from "moment";
-import React, { useContext, useState } from "react";
+import React, { useContext, useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/app/components/stardust/AssociatedOutput.tsx]: Resolved.

**Comment by msarcev** [comment]
image
I think it looks good.

One thing we could improve here, is the formatting inside the tooltip. Make it so that the text is not broken in weird places.
Wdyt ?

**Comment by abbasabid** [comment]
> image I think it looks good.
> 
> One thing we could improve here, is the formatting inside the tooltip. Make it so that the text is not broken in weird places. Wdyt ?

Done.

**Review by msarcev — APPROVED** [review]

**Pull request merged**